geany.appdata.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<component>
+  <id type="desktop">geany.desktop</id>
```

Comment:
It is recommended to match the .desktop file name, but it should be unique is all that is really required of it. From the doc for the appstream standard,

> The `<id>` tag is a short and unique identifier for this component. It must contain only ASCII characters, dots, hyphens and numbers, spaces are not allowed. Specialized metainfo types, such as application or fonts, may apply additional restrictions on the id tag value.
A general pattern for a valid ID tag is to use a reverse URL scheme, consisting of `<tld>.<vendor>.<product>.<type>`, e.g. `org.kde.gwenview.desktop` or `com.hugski.ColorHug2.firmware`.
>
> Note that the value of this tag must be unique across all distributions. In case it is not, distributors are expected to reject the conflicting components from inclusion into their metadata. 


 But I am not really sure if I understood the question completely. Please do clarify.